optuna/integration/tensorboard.py

```diff
@@ -0,0 +1,155 @@
+import optuna
+from optuna import type_checking
+import os
+
+if type_checking.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Dict  # NOQA
+    from typing import Optional  # NOQA
+    from typing import Tuple  # NOQA
+
+try:
+    import tensorboard
+    from tensorboard.plugins.hparams import api as hp
+    import tensorflow as tf
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+    hp = object
+    tensorboard = object
+    tensorflow = object
+
+
+def _check_tensorboard_availability() -> None:
+
+    if not _available:
+        raise ImportError(
+            "TensorBoard is not available. Please install TensorBoard to use this "
+            "feature. It can be installed by executing `$ pip install "
+            "tensorboard`. For further information, please refer to the installation guide "
+            "of TensorBoard. (The actual import error is as follows: " + str(_import_error) + ")"
+        )
+
+
+class TensorBoardCallback(object):
+    """Callback to track Optuna trials with TensorBoard.
+
+    This callback adds relevant information that is tracked by Optuna to TensorBoard.
+    Args:
+        dirname:
+            Directory to store TensorBoard logs.
+        param_distributions:
+            `dict` of parameters which should be saved to TensorBoard on each trial.
+            Entries in the dictionary have a form `param_name: (type, args)`, where
+            `param_name` is a `str` containing parameter name, type is `str` set to one of
+            `'uniform'`, `'loguniform'`, `'discrete_uniform'`, `'int_uniform'` or `'categorical'`,
+            and `args` is a tuple of parameters passed to respective constructor.
+        metric_name:
+            Name of the metric. Since the metric itself is just a number,
+            `metric_name` can be used to give it a name. So you know later
+            if it was roc-auc or accuracy.
+    """
+
+    def __init__(
+        self, dirname: "str", param_distributions: "Dict[str, Tuple[str, Any]]", metric_name: "str"
+    ) -> None:
+
+        _check_tensorboard_availability()
```

Comment:
By using DRY import, please replace it as follows.

```suggestion
        _imports.check()
```